application/src/main/java/io/redlink/smarti/webservice/LegacyConversationWebservice.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2017 Redlink GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.redlink.smarti.webservice;
+
+import io.redlink.smarti.api.QueryBuilder;
+import io.redlink.smarti.api.StoreService;
+import io.redlink.smarti.model.*;
+import io.redlink.smarti.model.config.ComponentConfiguration;
+import io.redlink.smarti.model.config.Configuration;
+import io.redlink.smarti.model.result.Result;
+import io.redlink.smarti.services.ClientService;
+import io.redlink.smarti.services.ConfigurationService;
+import io.redlink.smarti.services.ConversationService;
+import io.redlink.smarti.services.QueryBuilderService;
+import io.redlink.smarti.utils.ResponseEntities;
+import io.redlink.smarti.webservice.pojo.QueryUpdate;
+import io.redlink.smarti.webservice.pojo.TemplateResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+
+
+/**
+ *
+ */
+@CrossOrigin
+@RestController
+@RequestMapping(value = "legacy/conversation",
```

Comment:
see https://github.com/redlink-gmbh/smarti/pull/97#discussion_r141250011

Keeping the old web-service is a considerable additional effort, especially since we're also introducing changes in the data-model.